src/gtil/predict.cc

```diff
@@ -19,46 +19,103 @@
 #include "../threading_utils/parallel_for.h"
 #include "./pred_transform.h"
 
+namespace treelite {
+
+class GTILBridge {
+ public:
+  template <typename ThresholdType, typename LeafOutputType>
+  inline static const typename Tree<ThresholdType, LeafOutputType>::Node*
+  GetNode(const Tree<ThresholdType, LeafOutputType>& tree, int nid)  {
+    return &tree.nodes_[nid];
+  }
+};
+
+}  // namespace treelite
+
 namespace {
 
 using treelite::threading_utils::ThreadConfig;
 using PredTransformFuncType = std::size_t (*) (const treelite::Model&, const float*, float*);
 
-template <typename T>
-inline int NextNode(float fvalue, T threshold, treelite::Operator op,
-                    int left_child, int right_child, int default_child) {
-  if (std::isnan(fvalue)) {
-    return default_child;
+/*!
+ * \brief A dense feature vector.
+ */
+class FVec {
```

Comment:
When I was working on XGB, I find this problematic when data is extremely sparse.

Reply:
The current version of GTIL also suffers from the same problem.